data/rewards/calculator.go

```diff
@@ -84,10 +93,10 @@ func (calc *RewardCalculator) Calculate() (amt *balance.Amount, err error) {
 
 	// how much rewards left before year close
 	yearSupply := options.YearBlockRewardShares[year]
-	yearDistributed := calc.rewardYears.Years[year].Distributed
+	yearDistributed := calc.rewardYears.Years[year].TillLastCycle
```

Comment:
To clarify, you're persisting the amount accumulated in the last cycle to separate it from the current distribution. So basically the amount left for the year doesn't take into account the amount Distributed for the current cycle?

Reply:
> To clarify, you're persisting the amount accumulated in the last cycle to separate it from the current distribution. So basically the amount left for the year doesn't take into account the amount Distributed for the current cycle?

That's exactly what I am doing here.